File: skiCentar/skiCentar.Model/Lift.cs
using System.Collections.Generic;

namespace skiCentar.Model
{
    public class Lift
    {
        public int Id { get; set; }

        public string Name { get; set; } = null!;

        public int? LiftTypeId { get; set; }

        public int? Capacity { get; set; }

        public int? ResortId { get; set; }

        public bool? IsFunctional { get; set; }

        public string? StateMachine { get; set; }

        public virtual ICollection<LiftLocation> LiftLocations { get; } = new List<LiftLocation>();

        public virtual ICollection<LiftMaintenance> LiftMaintenances { get; } = new List<LiftMaintenance>();

        public virtual LiftType? LiftType { get; set; }

        public virtual Resort? Resort { get; set; }
    }
}

File: skiCentar/UI/skicentar_mobile/lib/models/lift.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:skicentar_mobile/models/lift-location.dart';
import 'package:skicentar_mobile/models/lift-type.dart';
import 'package:skicentar_mobile/models/resort.dart';

part 'lift.g.dart';
@JsonSerializable()
class Lift{
  int? id;
  String? name;
  bool? isFunctional;
  String? stateMachine;
  int? capacity;
  int? liftTypeId;
  int? resortId;
  Resort? resort;
  LiftType? liftType;
  List<LiftLocation>? liftLocations;
  
  Lift({this.id,this.name});

  factory Lift.fromJson(Map<String,dynamic> json) => _$LiftFromJson(json);

  Map<String,dynamic> toJson() => _$LiftToJson(this);
}